api/model/api.go

```diff
@@ -841,6 +841,17 @@ type APIDefinitionSpec struct {
 	EnableDetailedRecording *bool `json:"enable_detailed_recording,omitempty"`
 
 	GraphQL *GraphQLConfig `json:"graphql,omitempty"`
+
+	// AnalyticsPlugin is used to configure analytics plugin which enables editing or removal of all parts of analytics
+	// records, raw request and responses recorded by Tyk at the gateway level
+	// +optional
+	AnalyticsPlugin AnalyticsPluginConfig `bson:"analytics_plugin" json:"analytics_plugin,omitempty"`
```

Comment:
I think we do not need `bson` tags
```suggestion
	AnalyticsPlugin AnalyticsPluginConfig `json:"analytics_plugin,omitempty"`
```